Kudu.Contracts/Deployment/DeploymentInfoBase.cs

```diff
@@ -68,5 +68,9 @@ public bool IsValid()
         // won't update until after a process restart. Therefore, we copy the needed
         // files into a separate folders and run sync triggers from there.
         public string SyncFunctionsTriggersPath { get; set; } = null;
+
+        public string FileName { get; set; }
+
+        public bool DisableTouchingWatchedFile { get; set;}
```

Comment:
>DisableTouchingWatchedFile [](start = 20, length = 26)

Took a re-look at how this will be used. A better choice of name would be EnabledWatchedFile. That's because the use-case is to check for !DisableTouching (negation of a negation). Instead rename this EnabledWatchedFile and set it to true in the constructor (at the beginning of this fil)